### PR TITLE
Connect to api

### PR DESCRIPTION
**Pull request by MargaretTechworld**
In this PR:
- I made use of the Bookstore API as my backend.

- Fetched books from the Bookstore API after the component has rendered
- Refactored my add book and remove book features to persist changes in the server

Thank you for you review.

### CODE REVIEW

File: src/redux/books/BooksSlice.js [hunk 1]
```diff
@@ -1,44 +1,94 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+// import { addBook, removeBook, getBook } from '../api';
```
Comment (Davidosky007) [inline src/redux/books/BooksSlice.js]: * Kindly remove all commented code from your code base as it makes your codebase messy and unprofessional 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @MargaretTechworld 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* API works fine 👍 
* Well written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MargaretTechworld** [comment]
Thank you for your review. I have removed the commented code.

**Review by ademibowale — APPROVED** [review]
Hi @MargaretTechworld  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
- Fetched books from the Bookstore API✔️
- Make sure that useEffect is used when dispatching the fetchBooks function in <BookList> ✔️
- Verify that createAsyncThunk is imported in the project  ✔️
- Descriptive commit messages ✔️ 
- Good use of Github flow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**